cmd/manager/main.go

```diff
@@ -161,9 +164,40 @@ func main() {
 			},
 		}
 	}
+
+	// http/2 should be disabled due to its vulnerabilities. More specifically,
+	// disabling http/2 will prevent it from being vulnerable to the HTTP/2 Stream
+	// Cancellation and Rapid Reset CVEs. For more information see:
+	// - https://github.com/advisories/GHSA-qppj-fm5r-hxr3
+	// - https://github.com/advisories/GHSA-4374-p667-p6c8
+	disableHTTP2 := func(c *tls.Config) {
+		setupLog.Info("disabling http/2")
+		c.NextProtos = []string{"http/1.1"}
+	}
+
+	tlsOpts = append(tlsOpts, disableHTTP2)
+
+	metricsServerOptions := server.Options{
+		BindAddress:   metricsAddr,
+		SecureServing: true,
+		TLSOpts:       tlsOpts,
+
+		// FilterProvider is used to protect the metrics endpoint with authn/authz.
+		// These configurations ensure that only authorized users and service accounts
+		// can access the metrics endpoint.
+		FilterProvider: filters.WithAuthenticationAndAuthorization,
+
+		// Ensure that metrics is protected with certs managed by cert-manager
+		// If not informed, the metrics service provided by controller-runtime will generate
+		// and use its own self-assigned certs which is not recommended for production envs.
+		CertDir:  "/var/certs/",
+		CertName: "olm-ca.crt",
+		KeyName:  "ca.crt",
```

Comment:
These ought to be configured similarly to catalogd.

Reply:
Yes, we need to do the same here: https://github.com/operator-framework/catalogd/pull/460